library/firewall_lib.py

```diff
@@ -693,14 +739,19 @@ def main():
                 interface,
                 icmp_block,
                 set_default_zone,
+                firewalld_conf,
             )
         )
     ):
-        module.fail_json(
-            msg="One of service, port, source_port, forward_port, "
-            "masquerade, rich_rule, source, interface, icmp_block, "
-            "icmp_block_inversion, target, zone or set_default_zone needs to be set"
-        )
+        if firewalld_conf_emptied:
+            return module.exit_json(changed=False, __firewall_changed=False)
```

Comment:
It is the case where allow_zone_drifting is set on a version where it is unsupported

Reply:
I don't understand - `firewalld_conf_emptied` will only be true if `firewalld_conf` is set, but this code block cannot be reached if `firewalld_conf` is set?

Reply:
It can be, I have a unit test explicitly testing the ability for this block to be reached.
- User sets firewalld_conf with `allow_zone_drifting`
- on `check_firewalld_conf`, `allow_zone_drifting` is removed because it would cause idempotency issues otherwise on firewalld versions where `AllowZoneDrifting` no longer exists
- firewalld_conf_emptied == True, because firewalld_conf is now a length 0 dictionary, making it also falsy
- Upon reaching the conditional, firewalld_conf is both falsy and firewalld_conf_emptied == true, so if no other parameters are set, module.exit_json() will execute

Reply:
Though either way, if it needed explanation, I probably need a better way to do this.